### ISSUE TITLE
Plan for a11y

### ISSUE THREAD

**Issue by ckprice** [mozilla-services/screenshots#2251]
Ref: #2206

**Comment by ckprice** [comment]
I spoke to David Bolter on IRC, and he's going to see about having someone review this, and recommend an a11y MVP. I'll assign myself for now.

**Comment by yzen** [comment]
Hi here are some a11y issues that you need to consider:

Implementation:
* After activating the widget it is presented as a modal overlay. It is impossible for the keyboard or screen reader user to move focus into the overlay. It always remains on content. Expected behaviour: keyboard focus should move into overlay (meaningful element) and should never escape into content. All buttons, etc should be keyboard navigable.
* Content should be hidden from screen reader (and be not navigable by a keyboard) while interacting with the widget. Since it is partially opaque, you should probably use aria-hidden="true" on the content container while the widget is active.
* Post crop controls are also non-navigable with a keyboard.

Design:
* Cropping can only be done via mouse so it's essentially inaccessible to anyone using a screen reader or keyboard. This needs to be resolved at the design stage since it requires a lot of thought put into how cropping should be done with keyboard only.

PS Generally it is a good idea to try performing all the functions using a keyboard and no mouse. This is the lowest hanging fruit and is also most effective.

Cheers

**Comment by ianb** [comment]
@yzen: are there any tools you know of that do this freestyle region selection well using a keyboard?

**Comment by yzen** [comment]
I personally do not, but the interaction feels similar to drag and drop (sorry if it's not the case) which can be implemented in an accessible way (see https://dev.opera.com/articles/accessible-drag-and-drop/).

**Comment by ckprice** [comment]
We discussed this a bit, and would like to propose the following for MVP in 54:

- [ ] Buttons should be in the tab index
- [ ] User should be able to do a full screen grab with the keyboard
- [ ] Keyboard focus should be in the iframe

@yzen thoughts?

**Comment by ianb** [comment]
In 48556dfecbd688adc10cd5316a44cf7bb32faf74 I improved things some:

- Focus goes to the iframe when Screenshots is activated
- The three buttons each have tabindexes

After this, one tab goes to My Shots, two goes to Save Visible, three goes to Save Full Page.  After that it goes to browser chrome elements, and finally goes to the underlying document's elements.  E.g., if you go to google.com and hit tab enough, it will focus the search field and buttons.

Also the relevant elements were converted to buttons to allow enter/space to activate them.

Note we do NOT have a keyboard shortcut to actually start Screenshots.

**Comment by ckprice** [comment]
I reached out to dbolter and @yzen here and in email. We're going to move forward with this for an MVP for launch, feel free to re-open this/open a new issue for post-54 discussions.

**Comment by yzen** [comment]
Sorry to reply so late, yes the list of things for 54 looks good. Hopefully the remained of the issues does not slip though.
